src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs

```diff
@@ -878,11 +878,20 @@ public override void VisitCollectionElementInitializer(ICollectionElementInitial
         {
             Assert.Equal(OperationKind.CollectionElementInitializer, operation.Kind);
             var addMethod = operation.AddMethod;
+#pragma warning disable CS0612 // Type or member is obsolete
+            // Still making sure it doesn't crash
             var isDynamic = operation.IsDynamic;
```

Comment:
>  var isDynamic = operation.IsDynamic; [](start = 10, length = 38)

I continue to find this pattern confusing. If we are calling code to assert that it does not throw that is reasonable but I think we need to make it more first class. As is currently being written if I came across the code in a few years I may delete it because it looks unnecessary. Consider adding an assert method of the form

```
void AssertNoThrow(Action action);
```

I understand xunit removed this because of their principaled stance on the matter. At the same time though it makes the code far more readable and their stance simply doesn't account for cases like this.

Reply:
> Consider adding an assert method of the form ...

It feels like allocating a delegate just for this purpose is an overkill. Perhaps just a generic method that takes ```T```.

```
void AssertNoThrow<T>(T val) {}
```

---
In reply to: [187724762](https://github.com/dotnet/roslyn/pull/26781#discussion_r187724762) [](ancestors = 187724762)

Reply:
That works for me as well.

---
In reply to: [187731690](https://github.com/dotnet/roslyn/pull/26781#discussion_r187731690) [](ancestors = 187731690,187724762)

Reply:
Filed [https://github.com/dotnet/roslyn/issues/26848](https://github.com/dotnet/roslyn/issues/26848) to follow up on this, as it's well beyond the scope of this PR.

---
In reply to: [187733960](https://github.com/dotnet/roslyn/pull/26781#discussion_r187733960) [](ancestors = 187733960,187731690,187724762)